pkg/operator/operator.go

```diff
@@ -256,7 +259,56 @@ func New(logger logr.Logger, clientConfig *rest.Config, opts Options) (*Operator
 					},
 				}})
 		}),
-		CertDir: certDir,
+		TLSOpts: []func(*tls.Config){
+			func(c *tls.Config) {
+				var ca *x509.Certificate
+				rawCA, err := os.ReadFile(opts.ClientCAFile)
+				if err != nil {
+					if opts.CACert != "" {
+						var decErr error
+						rawCA, decErr = base64.StdEncoding.DecodeString(opts.CACert)
+						if decErr != nil {
+							logger.Error(decErr, "unable to decode CA Cert argument")
+							return
+						}
+					} else {
+						logger.Error(err, "unable to read client CA for webhook server, skipping")
+						return
+					}
+				}
+				ca, err = x509.ParseCertificate(rawCA)
+				if err != nil {
+					logger.Error(err, "unable to parse client CA for webhook server, skipping")
+					return
```

Comment:
Is it okay that we skipped? We could do error handling outside this method and just pass in the certificate here.

Reply:
Good idea. That ends up being a bit cleaner, I think.